### ISSUE TITLE
Move to Angular 4

### ISSUE THREAD

**Issue by pkozlowski-opensource** [ng-bootstrap/ng-bootstrap#1439]
There are several things that need to happen in order to fully move to Angular 4 (note: we do support Angular 4 already, this issue just summarizes things that need to happen to fully drop support to Angular 2):
* [x] cut a bug fix release just before (so people still on Angular 2.x can update)
* [x] change all the references from `<template>` to `<ng-template>` in src code
* [x] change all the references from `<template>` to `<ng-template>` in docs
* [x] update generated package.json constraints
* [x] update versions in documentation (demo, README etc.)
* [x] update plunker templates to use just one variable for ng versions (instead a separate one for core and a different one for a router)
* [x] add breaking change section to the release commit

**Comment by pkozlowski-opensource** [comment]
Note to self: most of the work is already done in #1415, we just need to pull individual commits and make sure that other remaining items are taken care of.

**Comment by alexnoise79** [comment]
+1

**Comment by mitulgolakiya** [comment]
+1

**Comment by FredLoney** [comment]
A work-around for now is to specify the 4.0 packages in `package.json`, create the `npm-shrinkwrap.json` similar to that shown below, rerun `npm install` and ignore the console `<template>` deprecation warning.

```
{
  "name": "myApp",
  "version": "my.version.number",
  "dependencies": {
    "@ng-bootstrap/ng-bootstrap": {
      "version": "1.0.0-alpha.23",
      "from": "node_modules/npm/@ng-bootstrap/ng-bootstrap@1.0.0-alpha.23",
      "dependencies": {
        "@angular/core": {
          "version": "4.0.2",
          "from": "@angular/core@2.3.1",
          "resolved": "file:jspm_packages/npm/@angular/core@4.0.2"
        },
        "@angular/common": {
          "version": "4.0.2",
          "from": "@angular/common@2.3.1",
          "resolved": "file:jspm_packages/npm/@angular/common@4.0.2"
        },
        "@angular/forms": {
          "version": "4.0.2",
          "from": "@angular/forms@2.3.1",
          "resolved": "file:jspm_packages/npm/@angular/forms@4.0.2"
        }
      }
    }
  }
}
```

**Comment by Serginho** [comment]
+1

**Comment by seevee** [comment]
+1

**Comment by tbowmo** [comment]
What is the timeframe of getting support for angular-4?

**Comment by seevee** [comment]
Based on #1510, looks like sometime next week.

**Comment by mendeza** [comment]
My team runs with Angular 4, no problems.

**Comment by Serginho** [comment]
@mendeza Tabs component breaks on angular 4.

**Comment by pkozlowski-opensource** [comment]
Just to be _very_ clear: we do _fully_ support Angular 4 today. This issue is only about switching our demo site and minimal peer dependency to Angular 4. It should have no noticeable impact on people using this project on Angular 4 already (apart from the removal of the warning for `<template>` tags). See more details in: https://github.com/ng-bootstrap/ng-bootstrap/issues/1337#issuecomment-288971917

I'm going to lock this issue as it starts to accumulating noise without adding signal.

@Serginho if you see _any_ problems with the latest release and Angular 4 please open a new issue following our issue template (plunker with a reproduce scenario is needed).

**Comment by pkozlowski-opensource** [comment]
This one just fixed by 26bc8df90eb0975c08f6d7592c387c3953849e9e - the alpha.25 release containing this fix will follow shortly.
